src/frontend/src/pages/TeamsPage/TeamMembersPageBlock.tsx

```diff
@@ -7,118 +7,157 @@ import { Autocomplete, Grid, IconButton, TextField } from '@mui/material';
-  const {
-    isLoading: setTeamMemberIsLoading,
-    isError: setTeamMemberIsError,
-    error: setTeamMemberError,
-    mutateAsync
-  } = useSetTeamMembers(team.teamId);
+  const { isLoading: allUsersIsLoading, isError: allUsersIsError, error: allUsersError, data: users } = useAllUsers();
+  const { isLoading: setTeamMembersIsLoading, mutateAsync: setTeamMembersMutateAsync } = useSetTeamMembers(team.teamId);
+  const { isLoading: setTeamHeadIsLoading, mutateAsync: setTeamHeadMutateAsync } = useSetTeamHead(team.teamId);
 
-  if (isError) return <ErrorPage message={error?.message} />;
-  if (setTeamMemberIsError) return <ErrorPage message={setTeamMemberError?.message} />;
-  if (isLoading || setTeamMemberIsLoading || !users) return <LoadingIndicator />;
+  const toast = useToast();
 
-  const handleSubmit = async () => {
-    try {
-      await mutateAsync(members.map((member) => member.id));
-      setIsEditingMembers(false);
-    } catch (error) {
-      alert(error);
-    }
-  };
+  if (allUsersIsError) return <ErrorPage message={allUsersError?.message} />;
+  if (allUsersIsLoading || setTeamMembersIsLoading || setTeamHeadIsLoading || !users) return <LoadingIndicator />;
 
   const hasPerms = auth.user && (isAdmin(auth.user.role) || auth.user.userId === team.head.userId);
 
-  const options = users
+  const memberOptions = users
     .filter((user) => user.userId !== team.head.userId && !team.leads.map((lead) => lead.userId).includes(user.userId))
     .sort((a, b) => (a.firstName > b.firstName ? 1 : -1))
     .map(userToAutocompleteOption);
 
-  const editButtons = (
-    <div style={{ display: 'flex' }}>
-      <NERFailButton
-        onClick={() => {
-          setIsEditingMembers(false);
-          setMembers(team.members.map(userToAutocompleteOption));
-        }}
-      >
-        Cancel
-      </NERFailButton>
-      <NERSuccessButton sx={{ ml: 2 }} onClick={handleSubmit}>
-        Save
-      </NERSuccessButton>
-    </div>
-  );
+  const headOptions = users
+    .filter((user) => memberOptions.some((option) => parseInt(option.id) === user.userId) && isHead(user.role))
+    .sort((a, b) => (a.firstName > b.firstName ? 1 : -1))
+    .map(userToAutocompleteOption);
+
+  const submitHead = async () => {
+    try {
+      await setTeamHeadMutateAsync(parseInt(head.id));
+      setIsEditingHead(false);
+    } catch (error) {
+      error instanceof Error && toast.error(error.message);
+    }
+  };
 
-  const editingView = (
-    <PageBlock title={'People'} headerRight={editButtons}>
-      <Grid container spacing={1}>
-        <Grid item xs={12}>
-          <DetailDisplay label="Head" content={fullNamePipe(team.head)} />
+  const submitMembers = async () => {
+    try {
+      await setTeamMembersMutateAsync(members.map((member) => parseInt(member.id)));
+      setIsEditingMembers(false);
+    } catch (error) {
+      error instanceof Error && toast.error(error.message);
+    }
+  };
+
+  const editingHeadView = (
```

Comment:
for all these helper components, write them as `EditingHeadsView = () => (</>)` and then you can call them as components `<EditingHeadsView/>` below